src/tbx/mod.rs

```diff
@@ -204,6 +204,8 @@ impl Reader {
     ///
     /// * `n_threads` - number of extra background reader threads to use
     pub fn set_threads(&mut self, n_threads: usize) -> Result<(), ThreadingError> {
+        assert!(n_threads > 0, "n_threads must be > 0");
```

Comment:
Am I right that we can simply use  `bcf::set_threads` here? Maybe move it to utils.rs before.

Reply:
We would need a common `ThreadingError` type then...

Reply:
Can we maybe go forward with this patch anyway? I don't know so much about Rust error types and the best practice here (in python it's considered best practice to have an `exceptions` module for a library). Also, I'm planning to add an `IndexedReader` for VCF/BCF soon anyway and we can some more cleanup afterwards and consolidate the shared things? E.g., we should be able to centralize all rust code for parallel bgzf into one place including the thread pool stuff I am planning to add.

Reply:
Ok!

Reply:
The most standard solution for error handling is using the error-chain crate. I want to migrate error handling to that, but I did not find the time so far. See https://docs.rs/error-chain